pyglidein/startd_cron_scripts/clsim_gpu_test.py

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env python
+
+import os
+from optparse import OptionParser
+from subprocess import check_output, STDOUT
+import xml.etree.ElementTree as ET
+
+OUTPUT_FILE = 'benchmark.xml'
+
+CLASSAD_MAP = {'I3CLSimModule_makeCLSimHits_makePhotons_clsim_AverageDeviceTimePerPhoton':
+               'PYGLIDEIN_METRIC_TIME_PER_PHOTON'}
+
+
+def main():
+
+    usage = "usage: %prog [options]"
+    parser = OptionParser(usage)
+    parser.add_option('-n', type='int', default=1,
+                      help="Number of simulations to run")
+    (options, args) = parser.parse_args()
+
+    try:
+        # This line is needed to access the nvidia gpus via OpenCL inside a container.
+        # TODO: Is this needed in a grid environment?
+        os.environ['LD_LIBRARY_PATH'] = ('/usr/local/nvidia/lib64:'
+                                         '{}'.format(os.environ['LD_LIBRARY_PATH']))
+
+        cmd = '{} {}'.format(os.path.join('/cvmfs/icecube.opensciencegrid.org/py2-v2',
+                                          os.environ['OS_ARCH'],
+                                          'metaprojects/simulation/V05-00-07/env-shell.sh'),
+                             os.path.join('/cvmfs/icecube.opensciencegrid.org/py2-v2',
+                                          os.environ['OS_ARCH'],
+                                          'metaprojects/simulation/V05-00-07/clsim/resources',
+                                          'scripts/benchmark.py'))
+        cmd += ' -n {}'.format(options.n)
+        check_output(cmd, shell=True, env=os.environ, stderr=STDOUT)
```

Comment:
Is there a reason why we do `shell=True`?  That's potentially a security hole, and also maybe unnecessary.

Also, why `env=os.environ`?  That's the default anyway.

Reply:
Good call.  I switched to `shell=False` and cleaned up the commands.